unicode_data/src/build.rs

```diff
@@ -0,0 +1,234 @@
+// Copyright 2025 the Parley Authors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+//! Exposes functionality that allows for building ICU4X data providers.
+
+use icu_codepointtrie_builder::{CodePointTrieBuilder, CodePointTrieBuilderData};
+use icu_collections::codepointtrie::TrieType;
+use icu_locale::LocaleFallbacker;
+use icu_properties::{
+    CodePointMapData, CodePointSetData,
+    props::{
+        BidiClass, Emoji, ExtendedPictographic, GeneralCategory, GraphemeClusterBreak, LineBreak,
+        RegionalIndicator, Script, VariationSelector,
+    },
+};
+use icu_provider::prelude::icu_locale_core::locale;
+use icu_provider::prelude::*;
+use icu_provider_adapters::fork::ForkByMarkerProvider;
+use icu_provider_export::{
+    DataLocaleFamily, DeduplicationStrategy, ExportDriver, blob_exporter::BlobExporter,
+};
+use icu_provider_source::SourceDataProvider;
+
+use crate::*;
+
+extern crate alloc;
+
+fn pack(
+    script: Script,
+    gc: GeneralCategory,
+    gcb: GraphemeClusterBreak,
+    bidi: BidiClass,
+    is_emoji_or_pictographic: bool,
+    is_variation_selector: bool,
+    is_region_indicator: bool,
+    is_mandatory_linebreak: bool,
+) -> u32 {
+    let s = script.to_icu4c_value() as u32;
+    let gc = gc as u32;
+    let gcb = gcb.to_icu4c_value() as u32;
+    let bidi = unicode_to_unicode_bidi(bidi) as u32;
+
+    (s << Properties::SCRIPT_SHIFT)
+        | (gc << Properties::GC_SHIFT)
+        | (gcb << Properties::GCB_SHIFT)
+        | (bidi << Properties::BIDI_SHIFT)
+        | ((is_emoji_or_pictographic as u32) << Properties::IS_EMOJI_OR_PICTOGRAPH_SHIFT)
+        | ((is_variation_selector as u32) << Properties::IS_VARIATION_SELECTOR_SHIFT)
+        | ((is_region_indicator as u32) << Properties::IS_REGION_INDICATOR_SHIFT)
+        | ((is_mandatory_linebreak as u32) << Properties::IS_MANDATORY_LINE_BREAK_SHIFT)
+}
+
+fn unicode_to_unicode_bidi(bidi: BidiClass) -> unicode_bidi::BidiClass {
```

Comment:
if you enable `icu_properties/unicode_bidi`, you can get this directly from ICU4X.